### PR TITLE
New Extension: Retool

### PR DESCRIPTION
**Pull request by andre347**
## Description

New Raycast extension that allows you quickly search the Retool documentation, copy the URL and preview the documentation page.

## Screencast

retool-raycast1
retool-raycast2
retool-raycast3
retool-raycast4
retool-raycast5

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/retool/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useEffect, useMemo, useState } from "react";
+import algoliasearch from "algoliasearch/lite";
+
+const APPID = "T28YKFATPY";
+const APIKEY =
+  "YWY1ZGRlZGY5M2I4YTkxZDg3Y2FiOWYyODc5M2RkMGMxNzQ0NzM2NmU5ZjFjMDI0Y2M2YmFjNTVjYTUzNjJmZHRhZ0ZpbHRlcnM9KHByb2plY3Q6NTljMDkxZDk5MGRlMzMwMDEwNzZiMzI5KSwodmVyc2lvbjpub25lLHZlcnNpb246NjI0ZjE3YWI1NjYwYzEwOWQ5MDUwMjRlKSwoaGlkZGVuOm5vbmUsaGlkZGVuOmZhbHNlKSwoaW5kZXg6Q3VzdG9tUGFnZSxpbmRleDpQYWdlKQ==";
```
Comment (AndrewUsher) [inline extensions/retool/src/utils.ts:6]: This probably should be something the user passes in

Comment (andre347) [reply extensions/retool/src/utils.ts:6]: Why? I grabbed these keys from the browser console. They are the same for anyone who reads the Retool docs on the web. Asking people to add these keys would not be really user friendly imo.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪

Would it be possible to have some kind of initial state with some items?

Like the Tailwind extension, not that it has to be that many, but just so it doesn't feel that empty 🙂
CleanShot 2022-08-23 at 22 29 04@2x

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by andre347** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> Would it be possible to have some kind of initial state with some items?
> 
> Like the Tailwind extension, not that it has to be that many, but just so it doesn't feel that empty 🙂
> 
> CleanShot 2022-08-23 at 22 29 04@2x
> 
> Also done in Ansible
> 
> CleanShot 2022-08-23 at 22 37 42@2x
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

Thanks for the feedback! Finally got around implementing this. There's now initial content when the app loads. I have also added icons for each item in the list.

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**